client/service/src/lib.rs

```diff
@@ -371,14 +371,15 @@ pub struct RpcSession {
 	metadata: (),
 }
 
+// TODO: (dp) safe to remove?
```

Comment:
yes, this is essentially a wrapper over the pubsub stuff in jsonrpc and we need to something similar but it's useless as the metadata is `()` :)

Reply:
Removing it snowballs a bit. I'll leave the TODO for now.